clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -595,6 +595,8 @@ public void close() {
     @Override
     public Node leastLoadedNode(long now) {
         List<Node> nodes = this.metadataUpdater.fetchNodes();
+        if (nodes.isEmpty())
```

Comment:
Does `nodes.isEmpty()` always mean that an empty cluster is initialized? Is throwing this exception differentiates this case from the one where the `leastLoadedNode` method wasn't able to select the least loaded from a set of existing (but maybe disconnected) nodes?

Reply:
`int offset = this.randOffset.nextInt(nodes.size());` will fail if `nodes.isEmpty()` is true, but it's a more cryptic exception. So, we simply get a more useful error now.

Reply:
Ok, thanks for explaining.